src/alert-converter/alert-converter.service.ts

```diff
@@ -24,19 +36,110 @@ export class AlertConverterService {
+                                timeoutDuration: 0 //Can be specified when an timeout rule exists.
+                            },
+                        };
+                        break;
+
+                    case LogType.ERROR:
+                        log = {
+                            type: LogType.ERROR,
+                            time: date.getTime(),
+                            source: this.windowsExporterUrl,
+                            detector: this.prometheusUrl,
+                            message: alert.annotations.description,
+                            data: null,//Can be specified when an timeout rule exists.
+                        };
+                        break;
+
+                    case LogType.CB_OPEN:
+                        log = {
+                            type: LogType.CB_OPEN,
+                            time: date.getTime(),
+                            source: this.windowsExporterUrl,
+                            detector: this.prometheusUrl,
+                            message: alert.annotations.description,
+                            data: null, //Can be specified when an timeout rule exists.
+                        };
+                        break;
+
+                    default:
+                        log = {
+                            type: null,
+                            time: date.getTime(),
+                            source: this.windowsExporterUrl,
+                            detector: this.prometheusUrl,
+                            message: alert.annotations.description,
+                            data: null,
+                        };
+                }
                 messages.push(log);
             }
         }
         return messages;
     }
+
+    /**
+     * Searches a String for the occurrence of a LogType.
+     * 
+     * @param alertLabelsAlertname the String containing the LogType.
+     * @returns the LogType or null if none was found.
+     */
+    private getLogType(alertLabelsAlertname: string): LogType {
+
+        if (alertLabelsAlertname.toLowerCase().search('cpu') >= 0) {
+            return LogType.CPU
+        } else if (alertLabelsAlertname.toLowerCase().search('timeout') >= 0) {
+            return LogType.TIMEOUT
+        } else if (alertLabelsAlertname.toLowerCase().search('error') >= 0) {
+            return LogType.ERROR
+        } else if (alertLabelsAlertname.toLowerCase().search('cbopen') >= 0) {
+            return LogType.CB_OPEN
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Searches a String for CpuLoad data. 
+     * 
+     * @param stringContainingData the String containing the CpuLoad.
+     * @returns the value or zero if none was found.
+     */
+    private getCPULoadOfString(stringContainingData: string): number {
+
+        if (stringContainingData.search('VALUE') >= 0) {
+            let start = stringContainingData.search('VALUE') + 8;
+            let end = start + 7;
```

Comment:
schreib in ein kommentar, woher diese magic numbers 7 und 8 kommen.
der unwissende leser und du in spätestens fünf wochen wissen nämlich nicht, was die zu bedeuten haben.